islandora_solr_metadata.install

```diff
@@ -5,101 +5,112 @@
+    return t('No islandora_solr_metadata_fields table and/or islandora_solr_metadata_associations table found in the database; nothing to import');
+  }
+
+  // This table is going to be rebuilt as we create the new configs, so just get
+  // the data, dump it, and then reconfigure it.
+  $old_cmodel_assoc = $db->select('islandora_solr_metadata_cmodels', 'ismc')
+    ->fields('ismc', [])
+    ->execute()
+    ->fetchAll(PDO::FETCH_ASSOC);
+  $db->delete('islandora_solr_metadata_cmodels')
+    ->execute();
+  // Specifically in the case where this is being ported from Drupal 7,
+  // configurations no longer have integer identifiers, so this field needs to
+  // be replaced with the new version.
+  if ($schema->fieldExists('islandora_solr_metadata_cmodels', 'configuration_id')) {
+    $schema->dropPrimaryKey('islandora_solr_metadata_cmodels');
+    $schema->dropField('islandora_solr_metadata_cmodels', 'configuration_id');
+    $schema->addField('islandora_solr_metadata_cmodels',
+      'configuration_name',
+      [
         'type' => 'varchar',
         'length' => 255,
         'not null' => TRUE,
-        'description' => 'The name of the content model assigned to this configuration',
+        'description' => 'The name of the configuration which this field belongs to',
       ],
-    ],
-    'primary_key' => ['configuration_id', ['cmodel', 191]],
-  ];
-  return $schema;
+      [
+        'primary_key' => ['configuration_name', ['cmodel', 191]],
+      ]);
+  }
+
+  $associations = $db->select('islandora_solr_metadata_associations', 'isma')
+    ->fields('isma', [])
+    ->execute()
+    ->fetchAllAssoc('configuration_name');
+  $config = \Drupal::configFactory()->getEditable('islandora_solr_metadata.configs');
+  $completed = 0;
+  foreach ($associations as $configuration_name => $association) {
+    $fields = $db->select('islandora_solr_metadata_fields', 'ismf')
+      ->fields('ismf', [])
+      ->condition('configuration_id', $association->configuration_id)
+      ->execute()
+      ->fetchAll(PDO::FETCH_ASSOC);
+    // Add the association proper.
+    $new_assoc = IslandoraSolrMetadataConfig::getEmptyConfig();
+    foreach ($old_cmodel_assoc as $old_assoc) {
+      if ($old_assoc['configuration_id'] == $association['configuration_id']) {
+        $new_assoc['cmodel_associations'][] = $old_assoc['cmodel'];
+      }
+    }
+    $new_assoc['description']['description_field'] = $association['description_field'];
+    $new_assoc['description']['description_label'] = $association['description_label'];
+    $truncation_info = unserialize($association['description_data']);
+    $new_assoc['description']['truncation'] = $truncation_info['truncation'];
+    $config->set($association['configuration_name'], $new_assoc);
+    // Add the fields.
+    foreach ($fields as &$field) {
+      $field += unserialize($field['data']);
+      $new_field = IslandoraSolrMetadataFieldConfig::getEmptyField();
```

Comment:
Should probably grab the `\Drupal::service('islandora_solr_metadata.field_config')` service, instead of using the class explicitly.